extensions/ql-vscode/src/remote-queries/variant-analysis-manager.ts

```diff
@@ -31,6 +32,8 @@ export class VariantAnalysisManager extends DisposableObject implements VariantA
   private readonly variantAnalysisResultsManager: VariantAnalysisResultsManager;
   private readonly variantAnalyses = new Map<number, VariantAnalysis>();
   private readonly views = new Map<number, VariantAnalysisView>();
+  private static readonly maxConcurrentTasks = 3;
+  public queue: PQueue;
```

Comment:
I think it might be better to keep this private and expose a method that will queue a new variant analysis download. That way, we can change the implementation of this method without needing to change all consumers (even though that's currently only in two files).

We should also be able to make this `readonly` and create the queue inline, so something like this:

```suggestion
  private readonly queue = = new PQueue({ concurrency: VariantAnalysisManager.maxConcurrentTasks });
```

Reply:
Have made this private: https://github.com/github/vscode-codeql/pull/1599/commits/1f4e1f27ae622f5b8231ae30618cb4f8957f1751